src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -49,7 +50,11 @@ public Command parseCommand(String userInput) throws ParseException {
             return new AddCommandParser().parse(arguments);
 
         case DeleteCommand.COMMAND_WORD:
-            return new DeleteCommandParser().parse(arguments);
+            if (mainPanelName.equals(MainPanelName.Detail)) {
+                return new DeleteAttributeCommandParser().parse(arguments);
+            } else {
+                return new DeleteCommandParser().parse(arguments);
+            }
```

Comment:
This might be confusing, as `DeleteCommand` contains `DeleteAttributeCommand` logic here as well.

I think more appropriate solution would be
```java
if (commandWord.equals(AddCommand) && AddCommand.canExecuteAt()) {
    return new AddCommandParser().parse(arguments);
} ....
```
We need to change switch to if-else statements, let me know if you have any issues,

Reply:
Will push this to subsequent PR.